src/Workspaces/Core/Portable/EmbeddedLanguages/DateAndTime/LanguageServices/DateAndTimeEmbeddedLanguage.cs

```diff
@@ -29,9 +29,9 @@ public DateAndTimeEmbeddedLanguage(EmbeddedLanguageInfo info)
             Document document, int position, CancellationToken cancellationToken)
         {
             var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var token = root.FindToken(position);
+            var token = root.FindTokenOnLeftOfPosition(position);
```

Comment:
why?

Reply:
To get ```InterpolatedStringTextToken```. For ```ToString``` method it returns the same token.

Reply:
Seems weird. If I was after the token in a format clause, what token is returned?

Reply:
```root.FindToken(position)``` returns ```}``` for ```$"Text {date:$$}"``` and for ```$"Text {date:M$$}"```

Reply:
ok.  but what does it return for FindTokenOnLeftOfPosition?  i would expect it to return the `:` token then.  Is that not hte case?  If FindToken returns the `}` chec if that's the case (i.e. you're at the start of that token, and move back one token.  That seems like the simpler approach to take.

Reply:
```FindTokenOnLeftOfPosition``` returns ```:``` for ```$"Text {date:$$}"``` and ```:M``` for ```$"Text {date:$$}"```. That exactly what we need here.

Reply:
ok.  can you please doc in the code.  thanks :)